imager_images/tests.py

```diff
@@ -1,26 +1,868 @@
+        user_john = User.objects.get(username='john')
+        album1 = Album()
+        album1.user = user_john
+        album1.title = "Album Title"
+        album1.description = "An Album"
+        album1.published = "pvt"
+        album1.save()
+        the_album = Album.objects.all()[0]
+        self.assertEquals(the_album.title, "Album Title")
+        self.assertEquals(the_album.description, "An Album")
+        self.assertEquals(the_album.published, "pvt")
+        self.assertEquals(the_album.date_uploaded, datetime.date.today())
+        self.assertEquals(the_album.date_modified, datetime.date.today())
+
+    def test_album_cover(self):
+        user_john = User.objects.get(username='john')
+        photo1 = Photo()
+        photo1.user = user_john
+        photo1.image = THE_FILE
+        photo1.save()
+        album1 = Album()
+        album1.user = user_john
+        album1.cover = photo1
+        album1.save()
+        the_album = Album.objects.all()[0]
+        self.assertEquals(the_album.cover, photo1)
+
+
+class LibraryTestCase(TestCase):
+    def setUp(self):
+        self.client = Client()
+
+        self.user1 = UserFactory(username='johndoe')
+
+        photo1 = PhotoFactory()
+        photo1.user = self.user1
+        photo1.save()
+
+        photo2 = PhotoFactory()
+        photo2.user = self.user1
+        photo2.save()
+
+        photo3 = PhotoFactory()
+        photo3.user = self.user1
+        photo3.save()
+
+        photo4 = PhotoFactory()
+        photo4.user = self.user1
+        photo4.save()
+
+        photo5 = PhotoFactory()
+        photo5.user = self.user1
+        photo5.save()
+
+        album1 = Album(title='album1')
+        album1.user = self.user1
+        album1.cover = photo1
+        album1.save()
+
+        album2 = Album(title='album2')
+        album2.user = self.user1
+        album2.cover = photo2
+        album2.save()
+
+        album3 = Album(title='album3')
+        album3.description = 'album3'
+        album3.user = self.user1
+        album3.save()
+
+        album4 = Album(title='album4')
+        album4.description = 'album4'
+        album4.user = self.user1
+        album4.save()
+
+        photo1.albums.add(album1)
+        photo2.albums.add(album2)
+
+        photo2.albums.add(album3)
+        photo3.albums.add(album3)
+
+        thumb1 = get_thumbnail(photo1.image, '1000x1000')
+        thumb2 = get_thumbnail(photo2.image, '1000x1000')
+        thumb3 = get_thumbnail(photo3.image, '1000x1000')
+        thumb4 = get_thumbnail(photo4.image, '1000x1000')
+        thumb5 = get_thumbnail(photo5.image, '1000x1000')
+
+        self.thumb_user1_all = [thumb1.url, thumb2.url,
+                                thumb3.url, thumb4.url,
+                                thumb5.url]
+        self.thumb_user1_loose = [thumb4.url, thumb5.url]
+        self.thumb_user1_no_cover_album = [thumb2.url, thumb3.url]
+
+        self.client.login(username=self.user1.username, password=PASSWORD)
```

Comment:
wow.  That's seriously a record.  Way too many lines of code.  Make helper methods and break that shit up.